### PR TITLE
Revert Swift 6

### PR DESCRIPTION
**Pull request by cobexer**
Remove the added Swift 6 support to reduce the risk of regressions in 8.11

This reverts relevant changes introduced in:
* #30712
* #30849
* #30832

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by cobexer** [comment]
@bot-gradle test RfR

File: platforms/native/platform-native/src/main/java/org/gradle/language/swift/SwiftVersion.java [hunk 1]
```diff
@@ -29,14 +29,7 @@ public enum SwiftVersion {
...
      * @since 5.4
      */
-    SWIFT5(5),
-
-    /**
-     * Swift 6 major version.
-     *
-     * @since 8.11
-     */
-    SWIFT6(6);
```
[the remark this answers is not in the document]
Comment (cobexer) [reply platforms/native/platform-native/src/main/java/org/gradle/language/swift/SwiftVersion.java:39]: That may be an option, I don't know how it interacts with the other changes in this PR though

**Review by ljacomet — APPROVED** [review]
Assuming this covers all the tests, let's do it

We will sort out proper support in 8.12

**Comment by ljacomet** [comment]
Only failure seems to be performance related, let's try to merge anyway

**Pull request merged**